dotnet-desktop-guide/net/winforms/controls/how-to-add-an-event-handler.md

```diff
@@ -0,0 +1,116 @@
+
+01. Add a new button to the form and name it **Button1**.
+01. Change the **Properties** toolbox mode to **Events** by pressing the event button (:::image type="icon" source="../media/shared/visual-studio-events-button.png" border="false":::).
+01. Double-click the **Click** event to generate a handler. This action opens the code window and generates a blank `Button1_Click` method.
+01. Replace the method code with the previous code above.
+
+For more information about C# events, see [Events (C#)](/dotnet/csharp/programming-guide/events/)
+For more information about Visual Basic events, see [Events (Visual Basic)](/dotnet/visual-basic/programming-guide/language-features/events/)
+
+### Remove the handler
+
+The [Add a handler](#add-a-handler) section used some code to demonstrate adding a handler. That code also contained a call to remove a handler:
+
+:::code language="csharp" source="snippets/how-to-add-an-event-handler/cs/Form1.cs" id="RemoveHandler":::
+:::code language="vb" source="snippets/how-to-add-an-event-handler/vb/Form1.vb" id="RemoveHandler":::
+
+This syntax can be used to remove any event handler from any event.
+
+For more information about C# events, see [Events (C#)](/dotnet/csharp/programming-guide/events/)
+For more information about Visual Basic events, see [Events (Visual Basic)](/dotnet/visual-basic/programming-guide/language-features/events/)
+
+## How to use multiple events with the same handler
+
+With the Visual Studio Visual Designer's **Properties** toolbox, you can select the same handler already in use by a different event. Follow the directions in the [Set the handler](#set-the-handler) section to select an existing handler instead of creating a new one.
+
+In C#, the handler is attached to a control's event in the form's designer code, which changed through the Visual Designer. For more information about C# events, see [Events (C#)](/dotnet/csharp/programming-guide/events/)
+
+In Visual Basic, the handler is attached to a control's event in the form's code-behind file, where the event handler code is declared. Multiple `Handles` keywords can be added to the event handler code to use it with multiple events. The Visual Designer will generate the `Handles` keyword for you and add it to the event handler. However, you can easily do this yourself to any control's event and event handler, as long as the signature of the handler method matches the event. For more information about Visual Basic events, see [Events (Visual Basic)](/dotnet/visual-basic/programming-guide/language-features/events/)
+
+The following code is for Visual Basic. If you're viewing this article in C# language mode, you may not see the code below. This code demonstrates how the same method can be used as a handler for two different [`Button.Click`](xref:System.Windows.Forms.Control.Click) events:
+
+:::code language="vb" source="snippets/how-to-add-an-event-handler/vb/Form2.vb" id="MultipleHandlers":::
```

Comment:
I'm confused by the purpose of this section. Can we do something like the other snippets which have a C# & VB version? Also, it's showing up when the language I've selected is C#.

Reply:
I wonder if they changed something with the language switcher. they used to NEVER show up. I wonder if you don't have multiple language entries one-line-after-the-other if it just displays it? Anyhooo, the reason I have this section is because VB technically has two ways of creating handlers, declaring them with `AddHandler` which is the same as C#'s `+=` operator, and the `Handles` keyword which C# has no equivalent of. VB in Visual Studio uses the `Handles` version. 😦